### PR TITLE
Add EDITOR_UPLOAD_MEDIA_PAUSED analytics event

### LINKED ISSUE #6516: Add EDITOR_UPLOAD_MEDIA_PAUSED analytics event
As referenced in https://github.com/wordpress-mobile/WordPress-Android/pull/19878#discussion_r1442490754, we should not continue tracking `media_service_upload_failed` events when uploads are paused, as it would negatively influenced the metrics we plan to analyze.

We should verify both Android and iOS are set up to appropriately track the analytic events we desire. 

When the device is offline and a media upload is paused, we might consider tracking an event report as something like `EDITOR_UPLOAD_MEDIA_PAUSED` with the expectation that the upload would be retried.

### PR DESCRIPTION
**Pull request by derekblank**
When a device is offline and media uploads are ongoing, throw a "paused" analytics event instead of a failure.

Resolves:

* https://github.com/wordpress-mobile/gutenberg-mobile/issues/6516

-----

## To Test:

1. Create a post and upload an image
2. Turn off network connectivity (via airplane mode or similar)
3. Observe that paused event fires instead of failure

-----

## Regression Notes

1. Potential unintended areas of impact
Analytics events

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing in Tracks

3. What automated tests I added (or what prevented me from doing so)
N/A (analytics)

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/EditorMedia.kt [hunk 1]
```diff
@@ -299,14 +300,7 @@ class EditorMedia @Inject constructor(
...
     fun onMediaUploadPaused(listener: EditorMediaUploadListener, media: MediaModel, error: MediaError) = launch {
-        val properties: Map<String, Any?> = withContext(bgDispatcher) {
-            analyticsUtilsWrapper
-                .getMediaProperties(media.isVideo, null, media.filePath)
-                .also {
-                    it["error_type"] = error.type.name
-                }
-        }
-        analyticsTrackerWrapper.track(EDITOR_UPLOAD_MEDIA_FAILED, properties)
+        analyticsTrackerWrapper.track(EDITOR_UPLOAD_MEDIA_PAUSED)
```
Comment (dcalhoun) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/EditorMedia.kt]: What is the rationale for diverging from the error event and _not tracking_ `properties`? My presumption is that the properties provide _some_ value.

**Review by dcalhoun — COMMENTED** [review]
I did not perform any testing, but the implementation looks sound to me. We should very the event arrives as expected. We should also verify and register the event for both the WordPress and Jetpack app.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/EditorMedia.kt (hunk 1 above)
Comment (derekblank) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/EditorMedia.kt]: > What is the rationale for diverging from the error event and not tracking properties? My presumption is that the properties provide some value.

Good point. The initial rationale was that `properties` would relate more specifically to errors. After logging out the key/values pairs of `properties`, this is an example of what would be sent to the paused analytics event as properties:

```
ext: 
age_ms: 2526548679
bytes: 231174
mime: image/jpeg
megapixels: 0
error_type: CONNECTION_ERROR
```

This matches what is sent to `EDITOR_UPLOAD_MEDIA_FAILED`, naturally, so it's probably worth sending to the paused event too. I've updated `onMediaUploadPaused` to send `properties`. 👍

File: WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/EditorMedia.kt [hunk 2]
```diff
@@ -306,7 +307,8 @@ class EditorMedia @Inject constructor(
                     it["error_type"] = error.type.name
                 }
         }
-        analyticsTrackerWrapper.track(EDITOR_UPLOAD_MEDIA_FAILED, properties)
+
+        analyticsTrackerWrapper.track(EDITOR_UPLOAD_MEDIA_PAUSED, properties)
```
Comment (SiobhyB) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/EditorMedia.kt]: We could use the `AnalyticsUtils.trackWithSiteDetails` function, as done in [this open PR](https://github.com/wordpress-mobile/WordPress-Android/pull/19900/files#diff-811c66badbcf014e14cefac711b422c3d7675677f2534df88dd5397ddaa9a064), to ensure the blog ID and other site type details are sent along with this event.

**Review by dcalhoun — APPROVED** [review]
I verified the test plan succeeds. I also added the `site` to the `track` call, as suggested in https://github.com/wordpress-mobile/WordPress-Android/pull/19890#discussion_r1448806544. Well done! 🚀

**Pull request merged**